src/Sulu/Component/Webspace/Webspace.php

```diff
@@ -495,10 +495,14 @@ public function getTemplates()
      *
      * @param string $type
      * @param string $template
+     * @param string $parentTemplate
      */
-    public function addDefaultTemplate($type, $template)
+    public function addDefaultTemplate($type, $template, $parentTemplate)
     {
-        $this->defaultTemplates[$type] = $template;
+        $this->defaultTemplates[$template] = [
```

Comment:
This might not work as expected... If you have the same template as default for multiple parent-templates, this way it will only work for one of them. I think we have to keep it the other way, and make this a multidimensional array, something like this:

```php
$this->defaultTemplates[$type][$parentTemplate] = $template;
```

This would only work if `null` is a valid array key, since `$parentTemplate` might be null... If that is not possible we would have to introduce a different structure. In general it would be favourable to introduce a proper object instead of an associative array.

Reply:
I created a class for it called DefaultTemplate, with attributes type, template and parentTemplate. This way we can add more features to it later on.

`$this->defaultTemplates[$type] = new DefaultTemplate($type, $template, $parentTemplate);`